app/models/pawn.rb

```diff
@@ -1,5 +1,76 @@
 class Pawn < Piece
 
+  def valid_move?(new_x, new_y)
+    on_board?(new_x, new_y) && passes_pawn_rules?(new_x, new_y)
+  end
+
+  def passes_pawn_rules?(x, y)
+    if two_squares_forward?(x, y)
+      return true
+    elsif one_square_forward?(x, y)
+      return true
+    elsif diagonal_move?(x, y)
+      return true
+    end
+
+    return false
+  end
+
+  def two_squares_forward?(x, y)
+    first_move_position? && not_backwards?(x, y) && (self.y_cord - y).abs == 2 && self.x_cord - x == 0
+  end
+
+  def one_square_forward?(x, y)
+    not_backwards?(x, y) && (self.y_cord - y).abs == 1 && self.x_cord - x == 0
+  end
+
+  def diagonal_move?(x, y)
+    diagonal_allowed?(x, y) && not_backwards?(x, y) && (self.y_cord - y).abs == 1 && (self.x_cord - x).abs == 1
```

Comment:
The thinking with absolute values was that for the white pieces, pawns have to move in a positive way (ie, from y=1 to y=2), but black needs to move in a negative way (from y=6 to y=5). How then do we account for the difference in direction dependent on color?